test/e2e/downwardapi_volume.go

```diff
@@ -92,6 +100,168 @@ var _ = Describe("Downward API volume", func() {
+		pod := &api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name:   podName,
+				Labels: labels,
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:    "client-container",
+						Image:   "gcr.io/google_containers/busybox",
+						Command: []string{"sh", "-c", "while true; do if [[ -e /etc/labels ]]; then cat  /etc/labels; fi; sleep 1; done"},
+						VolumeMounts: []api.VolumeMount{
+							{
+								Name:      "podinfo",
+								MountPath: "/etc",
+								ReadOnly:  false,
+							},
+						},
+					},
+				},
+				Volumes: []api.Volume{
+					{
+						Name: "podinfo",
+						VolumeSource: api.VolumeSource{
+							DownwardAPI: &api.DownwardAPIVolumeSource{
+								Items: []api.DownwardAPIVolumeFile{
+									{
+										Path: "labels",
+										FieldRef: api.ObjectFieldSelector{
+											APIVersion: "v1",
+											FieldPath:  "metadata.labels",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				RestartPolicy: api.RestartPolicyNever,
+			},
+		}
+
+		containerName := "client-container"
+		defer func() {
+			By("Deleting the pod")
+			f.Client.Pods(f.Namespace.Name).Delete(pod.Name, api.NewDeleteOptions(0))
+		}()
+		By("Creating the pod")
+		if _, err := f.Client.Pods(f.Namespace.Name).Create(pod); err != nil {
+			Failf("Failed to create pod: %v", err)
+		}
+		expectNoError(waitForPodRunningInNamespace(f.Client, pod.Name, f.Namespace.Name))
+
+		By("Sleeping to give kublet time to update volume")
+		time.Sleep(5 * time.Second)
+		podStatus, err := f.Client.Pods(f.Namespace.Name).Get(pod.Name)
+		if err != nil {
+			Failf("Failed to get pod status: %v", err)
+		}
+		nodeName := podStatus.Spec.NodeName
+		testContainerOutputMatcherNoPodInit("downward API volume plugin", f.Client, pod,
+			containerName, nodeName, f.Namespace.Name,
+			downwardapiVolumeMapFormat(pod.Labels), ContainSubstring)
+
+		//modify labels
+		podStatus.Labels["key3"] = "value3"
+		if _, err := f.Client.Pods(f.Namespace.Name).Update(podStatus); err != nil {
+			Failf("Failed to create pod: %v", err)
+		}
+		By("Sleeping to give kublet time to update volume")
+		time.Sleep(5 * time.Second)
+		testContainerOutputMatcherNoPodInit("downward API volume plugin", f.Client, pod,
+			containerName, nodeName, f.Namespace.Name,
+			downwardapiVolumeMapFormat(podStatus.Labels), ContainSubstring)
+	})
+
+	It("should verify update on annotations modification", func() {
+		annotations := map[string]string{}
+		annotations["builder"] = "bar"
+		podName := "annotationupdate" + string(util.NewUUID())
+		pod := &api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name:        podName,
+				Annotations: annotations,
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:    "client-container",
+						Image:   "gcr.io/google_containers/busybox",
+						Command: []string{"sh", "-c", "while true; do if [[ -e /etc/annotations ]]; then cat /etc/annotations; fi; sleep 1; done"},
```

Comment:
The mounttest image now has `--file_content_in_loop`, which I think it what you want.